freqtrade/persistence/models.py

```diff
@@ -259,16 +263,40 @@ class LocalTrade():
     strategy: str = ''
     timeframe: Optional[int] = None
 
-    def __init__(self, **kwargs):
-        for key in kwargs:
-            setattr(self, key, kwargs[key])
-        self.recalc_open_trade_value()
+    # Margin trading properties
+    borrowed_currency: str = None
+    collateral_currency: str = None
+    interest_rate: float = 0.0
+    liquidation_price: float = None
+    is_short: bool = False
+    __leverage: float = 1.0  # * You probably want to use self.leverage instead |
+    __borrowed: float = 0.0  # * You probably want to use self.borrowed instead V
 
-    def __repr__(self):
-        open_since = self.open_date.strftime(DATETIME_PRINT_FORMAT) if self.is_open else 'closed'
+    @property
+    def leverage(self) -> float:
+        return self.__leverage or 1.0
 
-        return (f'Trade(id={self.id}, pair={self.pair}, amount={self.amount:.8f}, '
-                f'open_rate={self.open_rate:.8f}, open_since={open_since})')
+    @property
+    def borrowed(self) -> float:
+        return self.__borrowed or 0.0
+
+    @leverage.setter
+    def leverage(self, lev: float):
+        if self.is_short is None or self.amount is None:
+            raise OperationalException(
+                'LocalTrade.amount and LocalTrade.is_short must be assigned before LocalTrade.leverage')
+        self.__leverage = lev
+        self.__borrowed = self.amount * (lev-1)
+        self.amount = self.amount * lev
+
+    @borrowed.setter
+    def borrowed(self, bor: float):
+        if not self.amount:
```

Comment:
Amounts should always be assigned (it's a mandatory field) when the trade object is created (look at `freqtradebot.py` and `backtesting.py` - these are the only places we create orders).

the only case where this "might" not be the case is tests - in which case the test should be fixed (this is easy to overlook in tests sometimes ...)

Reply:
Yeah it's just a safety check in case someone writes some sloppy code. This isn't in there anymore though, only leverage is a computed property. This is because for shorts, you actually can't determine the leverage based on the borrowed amount, and the amount, as the full amount will be borrowed